add references for LSST

ismael-mendoza opened the pull request.

aboucaud commented on galcheat/data/LSST.yaml:
```diff
@@ -59,23 +59,23 @@ references:
-    comment: ""
+    link: "https://arxiv.org/pdf/0805.2366v4.pdf"
+    comment: "See Table 2, first row"
   exposure_time:
-    link: "https://www.lsst.org/about/camera/features"
-    comment: "See Technical details"
+    link: "https://arxiv.org/pdf/0805.2366v4.pdf"
+    comment: "See Table 1 'Mean number of visits' and in the paper it is mentioned that visit exposure time 30s. So for example, for u band, 56 * 30 = 1680."
```
Here also, is there a reason to prefer the arXiv ref over the project official website ?
I have no opinion. 

@mpaillassa ?

mpaillassa replied:
Hm yes I tend to favor arXiv references but this is subjective and we may discuss about this.
But in this case `exposure_time` is in the arXiv reference but I can't find it in the [link](https://www.lsst.org/about/camera/features).

aboucaud replied:
In `Camera Data Rates` it says _15 sec integration_. But for this one I guess it's best if we refer directly to a paper stating the exposure time.

ismael-mendoza replied:
We can discuss in the meeting tomorrow, but I think I prefer referring the paper directly as well.

aboucaud commented on galcheat/data/LSST.yaml:
```diff
@@ -59,23 +59,23 @@ references:
   obscuration:
-    link: "https://arxiv.org/abs/0805.2366"
-    comment: "See https://github.com/aboucaud/galcheat/issues/38 for discussions"
+    link: "https://arxiv.org/abs/0805.2366v4.pdf"
```
At several places you decided to change the link from the arXiv ref to the actual pdf version.
Is there a specific reason ?

On the one hand the paper version "fixes" the value it references, which is good.
One the other hand, when looking at https://arxiv.org/abs/0805.2366 one can see there exists a much recent version of the paper (v5) and thus I'm left wondering why you chose the v4 instead.

mpaillassa replied:
For my case I found that most arXiv references at first were pdf links so I have put pdf links everywhere for consistency. Maybe @ismael-mendoza have thought the same ?

About the version it may be that we took it from WLD and that we did not double check it...

ismael-mendoza replied:
Yes I agree that we should have abstract pages in case there is a newer version, but I also like "fixing" the value of the reference. How about the url to the abstract page, but in the comment we specify which version we took values from ?

aboucaud replied:
Sounds about right :)

ismael-mendoza replied:
excellent I can push that change

ismael-mendoza commented:
I just pushed the change, I decided to follow the numbers in table 5 since they are more recent. So some of the values (mean_sky_level and psf_fwhm) got updated and the sky_level tests are not passing anymore. Should I keep the old values or update the tests ? 

Also (this might be a separate PR) I realized we should probably follow standard scientific convention in the values when possible. For example, obscuratoin currently reads `0.39257227` but I suspect the mirror diameter and the effetive one don't have enough decimals to actually report this many ? What do you think @mpaillassa @aboucaud ?

ismael-mendoza commented:
> That said, we have to be careful about the consistency across surveys, for instance for the psf_fwhm you decided to keep only two significant digits, but I'm not sure that's consistent.

Yes I decided to keep two significant digits because in the table 2 in the reference for v5 there are two significant digits.

mpaillassa commented:
About arXiv references, I created an issue #104 to review it for other survey references as well.

About significant digits, I did not really know for `obscuration` so I was quite conservative but I guess it is useless having that many digits. We may set a rule specifying the number of digits of each parameter to be consistent across surveys ?

aboucaud commented:
> I just updated the value of the tests so that they are now passing. I renamed `BTK_MEAN_SKY_LEVEL` to `MEAN_SKY_LEVEL` since these numbers are not technically from WLD/BTK anymore since they used the old version of the paper.

Well, if you used `galcheat` to compute `galcheat` values, that looses the interest of checking against an external value.
Would it be possible for you to modify the sky_brightness in WLD for LSST and then compute the floating point values ? 🙏🏼 
 
> > About significant digits, I did not really know for obscuration so I was quite conservative but I guess it is useless having that many digits. We may set a rule specifying the number of digits of each parameter to be consistent across surveys ?
> 
> That sounds good @mpaillassa, @aboucaud do you think we can merge this and create a separate issue for the significant digits?

Right, please do create a separate issue.

ismael-mendoza commented:
Hi @aboucaud - I recomputed the numbers in the test using WLD. I changed the sky brightness to be the one in galcheat (it was mostly the same) but the fluxes don't match exactly. 

The other difference might come from the zeropoints when we converted from ab magnitudes to classical magnitudes? The numbers are close though (< 5% difference), should we dig into this more or are you satisfied with them being approximately the same to the WLD ones?

aboucaud commented:
Quick question : did you try running the `check_zeropoints.py` script to make sure the zeropoint does not need an update ?

aboucaud commented on tests/test_utilities.py:
```diff
@@ -36,44 +37,48 @@ def lsst_btk_counts(request):
     survey, filt, expected = lsst_btk_sky_level
     sky_level = mean_sky_level(survey, filt).value
-    assert int(sky_level) == expected
+    np.testing.assert_allclose(int(sky_level), expected, rtol=1e-1)
```
For all the mean_sky_level tests, if you are testing relative tolerance, then the cast to integer should be removed from the test.

ismael-mendoza replied:
sounds good, I just removed them.

ismael-mendoza commented:
> Quick question : did you try running the check_zeropoints.py script to make sure the zeropoint does not need an update ?

Are you referring to updating the zeropoints in galcheat or comparing with WLD ? I just ran `check_zerpoints.py` and the zeropoints are the same as the ones in `data/LSST.yml`

aboucaud reviewed (approved):
Thanks a lot @ismael-mendoza !
